### PR TITLE
Add HTTP-Compression BBE

### PR DESCRIPTION
**Pull request by chamil321**
## Automation tests
 - Unit tests 
   > Added
 - Integration tests
   > N/A

### CODE REVIEW

File: examples/http-compression/tests/http_compression.bal [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import ballerina/test;
+import ballerina/io;
```
Comment (VijithaEkanayake) [inline examples/http-compression/tests/http_compression.bal]: Shall we reorder the imports in an alphabetical order?

File: examples/http-compression/http_compression.bal [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                    error e => log:printError("Error sending response", err = e) };
+            }
+        }
+    }
+}
+
+// The compression behaviour of service is inferred by its default value that is COMPRESSION_AUTO
+service<http:Service> backend bind listenerEndpoint {
+    echo(endpoint caller, http:Request req) {
+        string value;
```
Comment (VijithaEkanayake) [inline examples/http-compression/http_compression.bal]: Do we need to declare value variable here? Can't we declare and initialize in line 92?

File: examples/http-compression/tests/http_compression.bal [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import ballerina/test;
```
Comment (samgnaniah) [inline examples/http-compression/tests/http_compression.bal]: Shall we add some comments to this file as well?

File: examples/http-compression/http_compression.bal [hunk 4]
```diff
@@ -0,0 +1,98 @@
+import ballerina/http;
+import ballerina/log;
+
+endpoint http:Listener listenerEndpoint {
+    port: 9090
+};
+
+// Since compression behaviour of the service is set as COMPRESSION_AUTO, entity body compression is done according
```
Comment (samgnaniah) [inline examples/http-compression/http_compression.bal]: Let's use single backticks for "COMPRESSION_AUTO". Apply elsewhere for similar settings that you might do.

File: examples/http-compression/http_compression.bal [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+// to the scheme indicated in `Accept-Encoding` request header. When the header is not present or the
```
Comment (samgnaniah) [inline examples/http-compression/http_compression.bal]: Shall we reword this to "Compression is not performed when the header is not present or the header value is "identity"."

File: examples/http-compression/http_compression.bal [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        path: "/"
+    }
+    invokeEndpoint(endpoint caller, http:Request req) {
+        caller->respond({ "Type": "Auto compression" }) but {
+            error e => log:printError("Error sending response", err = e)
+        };
+    }
+}
+
+// COMPRESSION_ALWAYS will guarantee a compressed response entity body. Compression scheme is set to the
```
Comment (samgnaniah) [inline examples/http-compression/http_compression.bal]: Avoid future tense. The word "will" usually indicates that the sentence is in future tense. Instead say "COMPRESSION_ALWAYS guarantees a compressed..."

File: examples/http-compression/http_compression.client.out [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+>
+< HTTP/1.1 200 OK
+< content-type: application/json
+< content-length: 49
+<
+* Connection #0 to host localhost left intact
+{"Type":"Always but constrained by content-type"}
+
+
+# To invoke the alwaysCompress Service, use the following curl command.
```
Comment (samgnaniah) [inline examples/http-compression/http_compression.client.out]: Shall we say the Accept-Encoding header is specified in this case?

File: examples/http-compression/http_compression.bal [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+            }
+            error responseError => {
+                caller->respond({ "error": "error occurred while invoking the service" }) but {
+                    error e => log:printError("Error sending response", err = e) };
+            }
+        }
+    }
+}
+
+// The compression behaviour of service is inferred by its default value that is COMPRESSION_AUTO
```
Comment (samgnaniah) [inline examples/http-compression/http_compression.bal]: Shall we reword this to something like "The compression behaviour of the service is inferred by `COMPRESSION_AUTO`, which is the default value of the service."

File: examples/http-compression/http_compression.client.out (hunk 7 above)
Comment (chamil321) [reply examples/http-compression/http_compression.client.out]: Here the `Accept-Encoding` header is not specified. Therefore added it
